src/main/scala/com/wavesplatform/network/RxExtensionLoader.scala

```diff
@@ -162,11 +163,9 @@ object RxExtensionLoader extends ScorexLogging {
           log.error("Error applying extension", err)
           Left(GenericError(err))
         })
-        .flatMap { ar =>
-          Task {
-            s = onExtensionApplied(s, extensionBlocks, ch, ar)
-          }.executeOn(scheduler)
-        }.logErr.runAsync(scheduler)
+        .map { ar => s = onExtensionApplied(s, extensionBlocks, ch, ar) }
```

Comment:
When you do `map`, task gets executed in the same scheduler, so we do need flatmap here!

Reply:
Already fixed with `asyncBoundary(scheduler)`, it's a wip :D